src/qmapshack/gis/db/CDBProject.h

```diff
@@ -159,6 +159,7 @@ class CDBProject : public IGisProject
     };
 
     Qt::CheckState checkState = Qt::Unchecked;
+    friend class CDatabaseConflict;
```

Comment:
Why friend? Just because `enum action_e` is private? Move the enum to public section.